sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/connections/workspace_connection_subtypes.py

```diff
@@ -0,0 +1,231 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# pylint: disable=protected-access
+
+from os import PathLike
+from pathlib import Path
+from typing import IO, Dict, Optional, Union
+
+from azure.ai.ml._restclient.v2023_06_01_preview.models import (
+    WorkspaceConnectionPropertiesV2BasicResource as RestWorkspaceConnection,
+)
+from azure.ai.ml._schema.workspace.connections.workspace_connection_subtypes import (
+    CognitiveServiceWorkspaceConnectionSchema,
+    CognitiveSearchWorkspaceConnectionSchema,
+    OpenAIWorkspaceConnectionSchema,
+)
+from azure.ai.ml._utils._experimental import experimental
+from azure.ai.ml.constants._common import (
+    BASE_PATH_CONTEXT_KEY,
+    PARAMS_OVERRIDE_KEY,
+    API_VERSION_KEY,
+    API_TYPE_KEY,
+    KIND_KEY,
+)
+from azure.ai.ml.entities._credentials import (
+    ApiKeyConfiguration,
+)
+from azure.ai.ml.entities._util import load_from_dict
+from .workspace_connection import WorkspaceConnection
+
+
+# Dev notes: Any new classes require modifying the elif chains in the following functions in the
+# WorkspaceConnection parent class: _from_rest_object, _get_entity_class_from_type, _get_schema_class_from_type
+@experimental
+class OpenAIWorkspaceConnection(WorkspaceConnection):
```

Comment:
Let's call this AzureOpenAIWorkspaceConnection. Native OpenAI doesn't care about `api_version` and `api_type` for example.

Reply:
Changed